sympy/core/add.py

```diff
@@ -595,10 +595,13 @@ def _eval_is_nonnegative(self):
         from sympy.core.exprtools import _monotonic_sign
         if not self.is_number:
             c, a = self.as_coeff_Add()
-            if not c.is_zero and a.is_nonnegative:
+            if not c.is_zero:
                 v = _monotonic_sign(a)
```

Comment:
No, as_coeff_Add should only extract the purely numeric (i.e., integer, rational, or float) term. I would focus on fixing the _monotonic_sign function, and understanding why we can't just use _monotonic_sign(self).